"index out of range" crash when reading bad input file

@henridf stumbled onto an example where `zq` crashed when he accidentally asked it to read a bad input file. The specific input file happened to be a prior revision of `zq`'s own changelog. I can repro this currently with `zq` version `280f671` and the attached input file.

```
# zq -version
Version: v0.9.0-18-g280f671

# zq CHANGELOG.txt | zq "*" -
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/brimsec/zq/zio/zngio.(*Reader).ReadPayload(0xc00007caf0, 0x1858320, 0x20be108, 0x0, 0xc0001e1a50, 0x100e388, 0x30)
	/Users/phil/work/zq/zio/zngio/reader.go:92 +0x22f
github.com/brimsec/zq/zio/zngio.(*Reader).Read(0xc00007caf0, 0xc00007caf0, 0x10000, 0xc000258000)
	/Users/phil/work/zq/zio/zngio/reader.go:67 +0x32
github.com/brimsec/zq/zio/detector.match(...)
	/Users/phil/work/zq/zio/detector/reader.go:58
github.com/brimsec/zq/zio/detector.NewReader(0x152b900, 0xc0001ecf30, 0xc0001878f0, 0xc0001ecf30, 0xc0000100d8, 0x0, 0x0)
	/Users/phil/work/zq/zio/detector/reader.go:21 +0x27a
main.(*Command).inputReaders(0xc00019e640, 0xc00000e090, 0x1, 0x1, 0x0, 0x0, 0xc00007a980, 0x40, 0x40)
	/Users/phil/work/zq/cmd/zq/zq.go:275 +0x3be
main.(*Command).Run(0xc00019e640, 0xc00000e090, 0x1, 0x1, 0x0, 0x0)
	/Users/phil/work/zq/cmd/zq/zq.go:198 +0x24a
github.com/mccanne/charm.(*instance).run(0xc00000e960, 0xc00000e090, 0x1, 0x1, 0x0, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/instance.go:53 +0x28c
github.com/mccanne/charm.(*Spec).ExecRoot(0x184ff20, 0xc00000e090, 0x1, 0x1, 0xc000157f78, 0x10074cf, 0xc00007e058, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/charm.go:77 +0x96
main.main()
	/Users/phil/work/zq/cmd/zq/main.go:9 +0x78
```

[CHANGELOG.txt](https://github.com/brimsec/zq/files/4471034/CHANGELOG.txt)

It's an elusive rascal, though. If you try this with the _current_ rev of `zq`'s `CHANGELOG.md`, you instead get the more appropriate `malformed input`.

--- a/zio/zngio/reader.go
+++ b/zio/zngio/reader.go
@@ -89,6 +89,9 @@ again:
 	}
 	// remove newline
 	line = bytes.TrimSpace(line)
+	if len(line) == 0 {
+		return nil, nil, ErrBadFormat
+	}
 	if line[0] == '#' {
 		b, err := r.parseDirective(line)
 		if err != nil {

